engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -156,7 +158,17 @@ public ValidatorImpl(ConstraintValidatorFactory constraintValidatorFactory,
 		this.validationOrderGenerator = validationOrderGenerator;
 		this.traversableResolver = traversableResolver;
 		this.validatorScopedContext = new ValidatorScopedContext( messageInterpolator, parameterNameProvider, clockProvider, temporalValidationTolerance, scriptEvaluatorFactory, failFast, traversableResolverResultCacheEnabled );
-		this.constraintValidatorInitializationContext = new HibernateConstraintValidatorInitializationContextImpl( validatorScopedContext.getScriptEvaluatorFactory(), validatorScopedContext.getClockProvider(), validatorScopedContext.getTemporalValidationTolerance() );
+		this.scriptEvaluatorFactory = validatorScopedContext.getScriptEvaluatorFactory();
+		this.constraintValidatorInitializationContext = new HibernateConstraintValidatorInitializationContextImpl( validatorScopedContext.getScriptEvaluatorFactory(), validatorScopedContext.getClockProvider(), validatorScopedContext.getTemporalValidationTolerance(), null );
+	}
+
+	public ValidatorImpl withDynamicConstraintValidatorInitializationPayload(Object dynamicPayload) {
```

Comment:
I think I would stick to as simple name. Something like `withConstraintValidatorDynamicPayload()`. The fact that it is related to the initialization can be removed I think. We don't really care (mostly because it can still be used at runtime by adding an attribute to the constraint validator).